# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender_api/src/display_item.rs
# Commit: aaed31bbe8f4
# Full Hash: aaed31bbe8f4e0a8892e632633c27033549eaf33
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2020-03-19 03:53:34
# Regressor Bug: 1614655
# File Overlap Count: 2
# Description:
#   Bug 1614655 - Part 2: Allow 1:n mapping of Gecko - WR display items r=jrmuizel
#   
#   This patch changes the underlying storage for WR display items in DisplayItemCache
#   from Vec<Option<CachedDisplayItem> to Vec<Vec<CachedDisplayItem>>.
#   This allows storing multiple WebRender display items for one Gecko display item.
# ==============================================================================

diff -r 0e791acac783 -r aaed31bbe8f4 gfx/wr/webrender_api/src/display_item.rs
--- a/gfx/wr/webrender_api/src/display_item.rs	Wed Mar 18 23:46:27 2020 +0000
+++ b/gfx/wr/webrender_api/src/display_item.rs	Wed Mar 18 23:47:05 2020 +0000
@@ -162,7 +162,8 @@
     PopStackingContext,
     PopAllShadows,
 
-    ReuseItem(ItemKey),
+    ReuseItems(ItemKey),
+    RetainedItems(ItemKey),
 }
 
 /// This is a "complete" version of the DisplayItem, with all implicit trailing
@@ -204,8 +205,6 @@
     PopReferenceFrame,
     PopStackingContext,
     PopAllShadows,
-
-    ReuseItem(ItemKey),
 }
 
 #[derive(Clone, Copy, Debug, Default, Deserialize, PartialEq, Serialize, PeekPoke)]
@@ -1502,7 +1501,8 @@
             DisplayItem::Rectangle(..) => "rectangle",
             DisplayItem::ScrollFrame(..) => "scroll_frame",
             DisplayItem::SetGradientStops => "set_gradient_stops",
-            DisplayItem::ReuseItem(..) => "reuse_item",
+            DisplayItem::ReuseItems(..) => "reuse_item",
+            DisplayItem::RetainedItems(..) => "retained_items",
             DisplayItem::StickyFrame(..) => "sticky_frame",
             DisplayItem::Text(..) => "text",
             DisplayItem::YuvImage(..) => "yuv_image",